### DISCUSSION TITLE
Getting faster matrix operations in more recent R versions

### DISCUSSION THREAD

**Discussion by ChristineStawitz-NOAA** [NOAA-FIMS/FIMS#238] [Show and tell]
@Andrea-Havron-NOAA and I found this StackOverflow thread which works on my PC to get the faster matrix libraries working in a more recent version of R: https://stackoverflow.com/questions/38090206/linking-intels-math-kernel-library-mkl-to-r-on-windows

This requires admin privileges unless you have a portable R install (e.g. under Documents or somewhere other than Program Files) in which case you might be able to get it to work.

Instead of editing the Rprofile.site file as described in step 6, I just ran `options(repos = c(CRAN="@CRAN@"))` in my R session once I followed the steps and reopened R.

More information can be found on this [GitHub repository](https://github.com/goepp/r-blas) and [blog post](https://www.r-bloggers.com/2017/11/why-is-r-slow-some-explanations-and-mklopenblas-setup-to-try-to-fix-this/), which includes speed comparisons.

**Comment by kellijohnson-NOAA** [comment]
@Bai-Li-NOAA I am putting this discussion on your radar incase you think that it would be good to think about doing this in our custom image? I am not sure if it would be possible but I do not see individuals wanting to change their installation of R but if R is set up for them and they don't have to do anything but can reap the benefits of a gain in speed it might be worth it. There is no need to do anything about it now because we have other higher priorities but we can include it in a list of pros for using Google Cloud Workstations.

**Reply by kellijohnson-NOAA** [reply]
@Bai-Li-NOAA just putting this on your radar again as it was recently brought up in the [TMB Google Group](https://groups.google.com/g/tmb-users/c/vgKn27bPRmc/m/ehta-OT_AAAJ?utm_medium=email&utm_source=footer).
